### DISCUSSION TITLE
Type hints?

### DISCUSSION THREAD

**Discussion by jorgeleo** [sebastienros/jint#1628] [Ideas]
My request is very simple. To add optional type hints like typescript. no need for type validation, and i know that is not in the javascript standard, but it will help to provide better intellisence when editing scripts with the likes of monaco

**Comment by lahma** [comment]
There's already a discussion touching the subject [here](https://github.com/sebastienros/jint/discussions/1244).

**Comment by jorgeleo** [comment]
Yes, but that discussion takes it too far. That discussion is intending a full Type Script implementation, and that, I think is out of the scope of Jint.

I only would like for Jint not to reject if type hints are included in the file, just ignore them.

`
var a : string = "123";
`
`
var b : DateTime = '1/1/2001';
`

Thank you for considering it.

**Reply by lahma** [reply]
Esprima, the parser behind Jint is a JavaScript parser, not a TypeScript parser. So you need to feed the the TS compilation results to it (and Jint).

**Reply by jorgeleo** [reply]
Right!

I wanted to skip the TS compilation step... maybe ignoring the type hints is something that can be added to esprima.net?

**Reply by lahma** [reply]
It's a JS parser, so I'm afraid not...

**Reply by jorgeleo** [reply]
:^(

It worth to ask
